File: solutions_2692487_0/Python/LewCPE/mote.py


def play(armin,motes):
    for i,mote in enumerate(motes):
        if mote < armin:
            armin += mote
        elif armin == 1:
            return len(motes)
        else:
            add_oper = 0
            del_oper = 1 + play(armin,motes[i+1:])
            while armin <= mote:
                armin += armin - 1
                add_oper += 1
            armin += mote
            add_oper += play(armin,motes[i+1:])
            return min(del_oper,add_oper)
            break
    return 0

def main():
    for nocase in xrange(1,int(raw_input())+1):
        armin_size, nomote = [int(c) for c in raw_input().split()]
        mote_list = [int(c) for c in raw_input().split()]
        mote_list.sort()
        oper = play(armin_size,mote_list)
        print "Case #%d: %d" % (nocase,oper)
        
if __name__ == "__main__":
    main()

File: solutions_2692487_0/C++/Claudson/mote.cc

#include <stdio.h>

#include <vector>

#include <algorithm>

using namespace std;

main() {


  int T;

  scanf("%d",&T);

  for (int t = 1; t <= T; t++) {
    vector <long long int> motes;
    long long int A;
    int N;

    scanf("%lld %d", &A,&N);

    for (int i = 0; i < N; i++) {
      long long int x;
      scanf("%lld", &x);
      motes.push_back(x);      
    }
    
    sort(motes.begin(),motes.end());

    int count = 0;
    
    int min = N;
    
    if (A != 1) {
      for (int i = 0; i < N; i++) {
	
	//printf("mote %d = %lld\n",i,motes[i]);
	
	while (motes[i] >= A)  {
	  // printf("%lld , %lld\n",A,motes[i]);
	  A = A + A - 1;
	  count++;
	}
	
	if (motes[i] < A) {
	  A += motes[i];
	}
	
	if (N-i + count -1 < min) {
	  min = N - i + count - 1;
	}
      }
    }
    printf("Case #%d: %d\n",t,min);

  }


}
